doc/python_api_reference_vDev.md

```diff
@@ -7793,6 +7838,101 @@ def to_unitary_matrix(
     """
 ```
 
+<a name="stim.PauliStringIterator"></a>
+```python
+# stim.PauliStringIterator
+
+# (at top-level in the stim module)
+class PauliStringIterator:
+    """Iterates over all possible pauli_strings of weight specfied by
+    min_weight and max_weight.
+
+    Returns:
+        An Iterable[stim.PauliString] that yields the requested Pauli
+            string.
+
+    Examples:
+        >>> import stim
+        >>> pauli_iter = stim.PauliString.iter_all(10, min_weight=2, max_weight=3)
+        >>> n = 0
+        >>> for pauli_string in pauli_iter:
+        ...     n += 1
+        >>> n
+        3645
+    """
+```
+
+<a name="stim.PauliStringIterator.__iter__"></a>
+```python
+# stim.PauliStringIterator.__iter__
+
+# (in class stim.PauliStringIterator)
+def __iter__(
+    self,
+) -> stim.PauliStringIterator:
+    """Returns an independent copy of the pauli string iterator.
+    """
+```
+
+<a name="stim.PauliStringIterator.__next__"></a>
+```python
+# stim.PauliStringIterator.__next__
+
+# (in class stim.PauliStringIterator)
+def __next__(
+    self,
+) -> stim.PauliString:
+    """Returns the next iterated pauli string.
+    """
+```
+
+<a name="stim.PauliStringIterator.next_qubit_permutation"></a>
+```python
+# stim.PauliStringIterator.next_qubit_permutation
+
+# (in class stim.PauliStringIterator)
+def next_qubit_permutation(
+    self,
+) -> object:
+    """Get the next permutation of qubit labels.
+
+    It's alot easier to test more complicated edge cases in python
+    which largely arise due to the algorithm for generating the next
+    permutation. The user should probably not interact with this function.
+
+    Returns:
+        next_perm: numpy boolean array which represents the next qubit permutation
+
+    Examples:
+        >>> import stim
+        >>> pauli_iter = stim.PauliString.iter_all(5, min_weight=3, max_weight=5)
+        >>> pauli_iter.next_qubit_permutation()
+        array([ True,  True, False,  True, False])
+    """
+```
+
+<a name="stim.PauliStringIterator.seed_iterator"></a>
+```python
+# stim.PauliStringIterator.seed_iterator
+
+# (in class stim.PauliStringIterator)
+def seed_iterator(
+    self,
+    arg0: object,
+) -> None:
+    """Seed the iterator with a given qubit pattern.
```

Comment:
Not clear to me what this means.

Reply:
I'm actually going to remove this function, after closer thought it's not super helpful. The idea was to be able to "seed" the iterator at a specific bit pattern which may be difficult to reach if w was large, but this can be tested on the C++ side more easily, and not sure if it's actually useful on the python side.

Reply:
Actually scratch that, I needed it for testing random starting points for long strings with higher weight. I will change the name to something more descriptive

Reply:
I renamed this to set_current_permutation.